### ISSUE TITLE
Fix for Facebook Login Issue

### ISSUE THREAD

**Issue by sudharsan1988** [kriasoft/react-starter-kit#935]
Steps to reproduce:
1. Login using facebook & first time, your data will be stored in the database and you will be redirected to http://localhost:3000/#_=_ 
2. Clear your browser history 
3. Try login again with facebook login that breaks the website with an error in src/core/passport.js
   
   **_TypeError: Converting circular structure to JSON_** or sometimes I get **_failed to serialize user into session passportjs_**

I fixed it by removing `done(null, users[0]);`  in passport.js line number: 85
and added this line 
`done(null, {
    id: profile.id,
    email: profile._json.email,
});`

Here is the gist, https://gist.github.com/sudharsan1988/44a57c46ad10eeafedad4c5e2af15c06/revisions

Is this the right way to fix this issue?

**Comment by Jose4gg** [comment]
I have the same problem as you, i tried yoir solution but it dosent work, yes the error of convertin circular structure to json is gone, but now it dosent create the user in the database.

**Comment by jazblueful** [comment]
The correct way to fix it is replacing:

`done(null, users[0]);`

with:
```
const user = users[0];
done(null, {
  id: user.id,
  email: user.email,
});
```

users[0] is circular JSON object returned by Sequelizer that JWT has problem serializing into payload. You need to convert it to Plain Object first before returning it.

**Comment by jorrit** [comment]
A fix is proposed in PR #1184.

**Comment by ulani** [comment]
@sudharsan1988   thank you very much for crating this issue! Unfortunately, we have close it due to inactivity. Feel free to re-open it or [join](https://discord.com/invite/2nKEnKq) our [Discord](https://discord.com/invite/2nKEnKq) channel for [discussion](https://github.com/kriasoft/react-starter-kit/discussions/1950).

NOTE: The `main` branch has been updated with React Starter Kit v2, using JAM-style architecture.
